addons/website/tests/test_ui.py

```diff
@@ -304,3 +304,15 @@ def test_18_website_snippets_menu_tabs(self):
 
     def test_19_website_page_options(self):
         self.start_tour("/?enable_editor=1", "website_page_options", login="admin")
+
+    def test_29_website_backend_menus_redirect(self):
+        Menu = self.env['ir.ui.menu']
+        menu_root = Menu.create({'name': 'Test Root'})
+        Menu.create({
+            'name': 'Test Child',
+            'parent_id': menu_root.id,
+            'action': 'ir.actions.act_window,%d' % (self.env.ref('base.open_module_tree').id,),
+        })
+        self.env.ref('base.user_admin').action_id = self.env.ref('base.menu_administration').id
+        self.assertFalse(menu_root.action, 'The test relies on the fact that the top menu has no action.')
```

Comment:
Nitpicking: Was confused by the assertion message, should not it be the opposite? "There is an action linked to the top menu which defeatss the purpose of the test" something like that? Otherwise you seem to say it is an error that the test relies on the fact the top menu has no action.

Reply:
I think I see what you mean.
I removed the first part and moved it in some `()` so it does not looks like it's part of the problem (why this line failed) but looks like it explain why it should be the case.
```suggestion
        self.assertFalse(menu_root.action, 'The top menu should not have an action (or the test/tour will not test anything).')
```
(To keep the same sentence, I should've added it as a python comment above the line and omit the string in the `assert`.)